lib/internal/crypto/random.js

```diff
@@ -119,6 +120,51 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 281474976710655;
+
+function randomInt(min, max, cb) {
+  if (typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!Number.isInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'integer', min);
+  }
+  if (!Number.isInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'integer', min);
+  }
+  if (!(max >= min)) {
+    throw new ERR_OUT_OF_RANGE('max', `>= ${min}`, max);
+  }
+  if (!(max <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE('max', `<= ${RAND_MAX}`, max);
+  }
+
+  // First we generate a random int between [0..bound)
+  const bound = max - min;
+  const range = bound - 1;
```

Comment:
I’m a bit confused by the `- 1` here. Shouldn’t we be using `bound` everywhere instead of `range`?

If `max` is 4 and `min` is 1, like in the example below, then `bound` is 3, `range` would be 2, and the result would be calculated as `x % 2 + 1` – which could never be 3, even though that is less than `max`. That seems a bit odd?

Reply:
Sorry about that, I opened the pull request too early, fixed in latest push.